src/sage/rings/function_field/drinfeld_modules/homset.py

```diff
@@ -421,3 +425,264 @@ def _element_constructor_(self, *args, **kwds):
+
+            ::
+
+            sage: Fq = GF(5)
+            sage: A.<T> = Fq[]
+            sage: K.<z> = Fq.extension(3)
+            sage: phi = DrinfeldModule(A, [z, 3*z, 4*z])
+            sage: chi = DrinfeldModule(A, [z, 2*z^2 + 3, 4*z^2 + 4*z])
+            sage: H = Hom(phi, chi)
+            sage: H.basis()
+            []
+
+        ALGORITHM:
+
+            We return the basis of the kernel of a matrix derived from the
+            constraint that `\iota \phi_T = \psi_T \iota` for any morphism
+            `iota`. 
+        """
+        domain, codomain = self.domain(), self.codomain()
+        Fq = domain._Fq
+        K = domain.base_over_constants_field()
+        q = Fq.cardinality()
+        char = Fq.characteristic()
+        r = domain.rank()
+        n = K.degree(Fq)
+        qorder = logb(q, char)
+        K_basis = [K.gen()**i for i in range(n)]
+        dom_coeffs = domain.coefficients(sparse=False)
+        cod_coeffs = codomain.coefficients(sparse=False)
+        # The commutative polynomial ring in tau^n.
+        poly_taun = PolynomialRing(Fq, 'taun')
+
+        sys = Matrix(poly_taun, n**2, n**2)
+
+        # Build a linear system over the commutative polynomial ring
+        # Fq[tau^n]. The kernel of this system consists of all
+        # morphisms from domain -> codomain.
+        for j in range(n):
+            for k in range(n):
+                for i in range(r+1):
+                    # Coefficients of tau^{i + k} coming from the
+                    # relation defining morphisms of Drinfeld modules
+                    # These are elements of K, expanded in terms of
+                    # K_basis.
+                    c_tik = (dom_coeffs[i].frobenius(qorder*k)*K_basis[j] \
+                            - cod_coeffs[i]*K_basis[j].frobenius(qorder*i)) \
+                            .polynomial().coefficients(sparse=False)
+                    c_tik += [0 for _ in range(n - len(c_tik))]
+                    colpos = k*n + j
+                    taudeg = i + k
+                    for b in range(n):
+                        sys[(taudeg % n)*n + b, colpos] += c_tik[b] * \
+                                poly_taun.gen()**(taudeg // n)
+
+        sol = sys.right_kernel().basis()
+        # Reconstruct basis as skew polynomials.
+        basis = []
+        tau = domain.ore_polring().gen()
+        for basis_vector in sol:
+            basis_poly = 0
+            for i in range(n):
+                for j in range(n):
+                    basis_poly += basis_vector[n*i + j].subs(tau**n)*K_basis[j]*tau**i
+            basis.append(self(basis_poly))
+        return basis
+
+    def _frobenius_matrix(self, order=1, K_basis=None):
+        r"""
+        Internal method for computing the matrix of the Frobenius endomorphism
+        for K/Fq. This is a useful method for computing morphism ring bases so
+        we provide a helper method here. This should probably be part of the
+        Finite field implementation.
```

Comment:
Agreed. Eventually, it will probably also will have to be implemented for Xavier's motives.